apps/web/src/components/proofreading-progress.tsx

```diff
@@ -182,97 +198,203 @@ export const ProofreadingProgress = ({
   const contributorsLength = proofreadingData.contributors.length;
 
   return (
-    <div
-      className={cn(
-        'group p-2.5 hover:p-5 rounded-[20px] shadow-course-navigation justify-start items-start gap-2.5 inline-flex absolute right-6 top-4',
-        'max-md:hidden', //proofreadingData.isOriginalLanguage ? 'hidden' : 'max-md:hidden',
-        mode === 'dark'
-          ? 'bg-newBlack-3'
-          : contributorsLength > 2
-            ? 'bg-brightGreen-1'
-            : 'bg-darkOrange-0',
-      )}
-    >
-      <div className="group-hover:hidden flex items-center gap-1.5">
-        <SmallProgressImage progress={contributorsLength} />
-        <span
-          className={cn(
-            'max-lg:hidden body-12px-medium w-[118px] whitespace-pre-line',
-            mode === 'dark' ? 'text-white' : 'text-black',
-          )}
-        >
-          {contributorsLength === 3
-            ? t('proofreading.completed')
-            : t('proofreading.inProgress')}
-        </span>
-      </div>
-      <div className="hidden group-hover:flex max-w-[258px] flex-col gap-4 justify-center">
-        <div className="flex items-center gap-4">
-          <span
+    <div>
+      <Dialog>
+        <DialogTrigger asChild>
+          <button
+            className="button-class absolute right-3 top-3 lg:hidden"
+            aria-label="Open Proofreading Progress Dialog"
+          >
+            <img src={ProgressMobile} alt="" />
+          </button>
+        </DialogTrigger>
+        <DialogPortal>
+          <DialogOverlay className="fixed inset-0 z-50 bg-black/80" />
+          <DialogContent
+            showCloseButton={false}
             className={cn(
-              'px-2 py-1 rounded-md shadow-course-navigation-sm title-medium-sb-18px',
+              'my-2 fixed left-1/2 top-1/2 z-50 grid -translate-x-1/2 -translate-y-1/2 gap-4 border-0 p-6 rounded-[20px] w-[280px]',
               mode === 'dark'
-                ? 'bg-white/15 text-newGray-4'
+                ? 'bg-newBlack-3'
                 : contributorsLength > 2
-                  ? 'bg-brightGreen-2 text-brightGreen-7'
-                  : 'bg-[#ff5c00]/15 text-darkOrange-6',
+                  ? 'bg-brightGreen-1'
+                  : 'bg-darkOrange-0',
             )}
           >
-            {contributorsLength}/3
-          </span>
+            <DialogClose className="flex justify-end rounded-sm focus:outline-none disabled:pointer-events-none data-[state=open]:bg-white data-[state=open]:text-muted-foreground">
+              <IoMdClose
+                className={cn(
+                  'size-6',
+                  mode === 'dark' ? 'text-newGray-2' : 'text-darkOrange-6',
+                )}
+              />
+              <span className="sr-only">Close</span>
+            </DialogClose>
+            <div className="flex flex-col relative pointer-events-none">
+              <LargeProgressImage progress={contributorsLength} />
```

Comment:
Images look too small on mobile compared to figma, we may need to resize them to remove their paddings